torch/onnx/_internal/exporter.py

```diff
@@ -1277,6 +1277,13 @@ def export(self) -> ONNXProgram:
                     "ONNXScript optimizer is not available. Skipping optimization. "
                     "Please `pip install onnxscript -U` to enable post-export optimization."
                 )
+            except Exception as e:
```

Comment:
When there is an exception, onnx_model may have been changed in place already and become invalid.

Reply:
Maybe not since it is the output of infer_shape; but I would confirm that to make sure

Reply:
Ok, what is the recommendation? Improve error message to let user know?

Reply:
I think the passes make a copy so we should be good. We may need to make the atomic expectation clear. Currently this is not explicit iirc. @gramalingam @BowenBao

Reply:
It is crucial that the optimization is atomic. Either we succeed with a modified graph or we fail and the input model is unchanged. My understanding was that it was atomic already, but if it is not, we really should make it so

Reply:
A good point to discuss in the optimizer implementation. It should be safe for the top-level function (`optimizer.optimize`) which does not modify the input model, but returns a new optimized model. It becomes tricky only if we try to eliminate this copy.

Reply:
Created https://github.com/microsoft/onnxscript/issues/1476